### PR TITLE
Added Texture manager

### PR DESCRIPTION
**Pull request by markJavab**
# Description

A resource manager that as for now only manages textures, audio sources will be added in a later stage. The class has been designed to support multiple resources. The benefit of the texture manager is to reduce memory usage by using only one instance of a resource instead of multiple of the same resource.

# How can this be tested?

Please describe the tests that you can run to verify your changes. Please also note any relevant details for your test configuration.

- Test for memory leaks with valgrind.
- Run the game and check if all the textures show up as normal.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Included 1 or more screenshots
- [x] Code is const correct

### CODE REVIEW

File: include/brickengine/resource_manager.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    ResourceManager(std::shared_ptr<Renderer> renderer);
+    std::shared_ptr<SDL_Texture> getTexture(std::string path);
+private:
+    std::shared_ptr<SDL_Texture> createTexture(std::string path) const;
+    std::unordered_map<std::string, std::weak_ptr<SDL_Texture>> texture_pool;
+
+    std::shared_ptr<Renderer> renderer;
+};
+
+#endif /* FILE_RESOURCE_MANAGER_HPP */
```
Comment (JanSchollaert) [inline include/brickengine/resource_manager.hpp]: Oh oh oh...

File: include/brickengine/engine.hpp [hunk 2]
```diff
@@ -41,8 +42,9 @@ class BrickEngine {
     int window_width;
     std::unique_ptr<Renderable> fps_counter;
     std::unique_ptr<SDL_Window, void(*)(SDL_Window*)> window;
-    std::shared_ptr<Renderer> renderer;
     std::unique_ptr<RenderableFactory> renderableFactory;
+    std::shared_ptr<Renderer> renderer;
+    std::shared_ptr<ResourceManager> resource_manager;
 };
 
 #endif /* FILE_BRICKENGINE_HPP */
```
Comment (Bram-Boris) [inline include/brickengine/engine.hpp]: @JanSchollaert it is also missing here :P

File: include/brickengine/rendering/renderable_factory.hpp [hunk 3]
```diff
@@ -21,7 +22,7 @@ class RenderableFactory{
     std::unique_ptr<Line> createLine(int x1, int y1, int x2, int y2, Color color, int layer) const;
 private:
     std::shared_ptr<Renderer> renderer;
-    SDL_Texture* createBMPTexture(std::string path) const;
+    std::shared_ptr<ResourceManager> resource_manager;
 };
 
 #endif
```
Comment (Bram-Boris) [inline include/brickengine/rendering/renderable_factory.hpp]: and here //

**Comment by Peter-JanGootzen** [comment]
What would we do without these incredible code reviews, lmao

**Pull request merged**